Packages/vcs/Lib/configurator.py

```diff
@@ -258,81 +199,124 @@ def detach(self):
+        Iterates all renderers, checks if there's an actor at the point
+        """
+        obj = None
+        layer_obj = 0
+
+        for ren in vtkIterate(self.render_window.GetRenderers()):
+            layer = ren.GetLayer()
+
+            if self.interactor is not None:
+                manager = vcs.vtk_ui.manager.get_manager(self.interactor)
+                if ren == manager.actor_renderer or ren == manager.renderer:
+                    continue
+
+            if obj is not None and layer < layer_obj:
+                continue
+
+            if self.picker.PickProp(x, y, ren):
+                obj = self.picker.GetViewProp()
+                layer_obj = layer
+
+        return obj
+
+
     def hover(self, object,event):
         if self.clicking is not None:
             return
 
         point = self.interactor.GetEventPosition()
-        cursor = self.interactor.GetRenderWindow().GetCurrentCursor()
-        if self.target:
-            if self.target.handle_click(point):
+
+        actor = self.actor_at_point(*point)
+        window = self.render_window
+
+        cursor = self.render_window.GetCurrentCursor()
+
+        new_cursor = vtk.VTK_CURSOR_DEFAULT
+        if actor is None:
+            man = vtk_ui.manager.get_manager(self.interactor)
+            if man.widget_at_point(*point):
+                new_cursor = vtk.VTK_CURSOR_HAND
+        else:
+            if self.target and self.target.handle_click(point):
                 if self.interactor.GetControlKey() == 1 and type(self.target) in (editors.marker.MarkerEditor, editors.text.TextEditor):
-                    if cursor != vtk.VTK_CURSOR_CROSSHAIR:
-                        self.interactor.GetRenderWindow().SetCurrentCursor(vtk.VTK_CURSOR_CROSSHAIR)
+                    new_cursor = vtk.VTK_CURSOR_CROSSHAIR
                 else:
-                    if cursor != vtk.VTK_CURSOR_HAND:
-                        self.interactor.GetRenderWindow().SetCurrentCursor(vtk.VTK_CURSOR_HAND)
-                return
-        else:
-            if self.toolbar.in_toolbar(*point):
-                return
+                    new_cursor = vtk.VTK_CURSOR_HAND
+            else:
+                display, key = self.display_and_key_for_actor(actor)
 
-        if self.marker_button.in_bounds(*point) or self.text_button.in_bounds(*point):
-            return
+                if display is not None and editable_type(display, key):
+                    new_cursor = vtk.VTK_CURSOR_HAND
 
-        for display in self.displays:
-            obj = self.in_display_plot(point, display)
-            if obj is not None:
-                if cursor != vtk.VTK_CURSOR_HAND:
-                    self.interactor.GetRenderWindow().SetCurrentCursor(vtk.VTK_CURSOR_HAND)
-                return
-        if cursor != vtk.VTK_CURSOR_DEFAULT:
-            self.interactor.GetRenderWindow().SetCurrentCursor(vtk.VTK_CURSOR_DEFAULT)
+        if cursor != new_cursor:
+            window.SetCurrentCursor(new_cursor)
```

Comment:
Ah, kk. I think most of why that was in there was because I was getting a ton of modified events at some point, and I wanted to reduce that number; I may have been repeatedly updating the cursor in every pass of this function at that point.